src/app/components/module-index-page/module-index-page.component.ts

```diff
@@ -1,193 +1,389 @@
-import { ToastrService } from 'ngx-toastr';
-import { globalCacheBusterNotifier } from 'ngx-cacheable';
 
 /** Typescript Component for Module Index Page */
 @Component({
-   selector: 'app-module-index-page',
-   templateUrl: './module-index-page.component.html',
-   styleUrls: ['./module-index-page.component.css']
+    selector: 'app-module-index-page',
+    templateUrl: './module-index-page.component.html',
+    styleUrls: ['./module-index-page.component.css']
 })
+
 export class ModuleIndexPageComponent implements OnInit {
 
-   /** Map of Visibility status of each Module */
-   contentVisible: Map<Module, boolean> = new Map<Module, boolean>();
-
-   /** Map of Modules to their list of related Content.
-    * Loaded when user clicks on Module (lazy load) */
-   moduleContents: Map<Module, Content[]> = new Map<Module, Content[]>();
-
-   /**
-    * Variable that will reference selected content for removal. Pre-initialized as it would 
-    * cause errors upon loading the component.
-    */
-   //Note that this needs defualt values so the bindings {{ }} in html will work on page load
-   selCon: Content = new Content(0, "", "", "", "", []);
-
-   /**
-    * Variable that will reference the module of the selected content for removal. 
-    * Pre-initialized as it would cause errors upon loading the component.
-    */
-   //Note that this needs defualt values so the bindings {{ }} in html will work on page load
-   selModule: Module = new Module(0, "", 0, [], null, null, null);
-
-   /** Used to display a spinner when modules are loading.*/
-   isLoading: boolean = false;
-
-   /** Create nodes to load child modules as objects */
-   nodes: any[] = this.ms.nodes;
-
-   /**
-    * Constructor for Module Index Component
-    * @param cs Fetches content
-    * @param ms Fetches tags
-    */
-   constructor(
-      public cs: ContentFetcherService,
-      public ms: ModuleStoreService,
-      private toastr: ToastrService,
-      private mfs: ModuleFetcherService
-   ) { }
-
-   /** On page initialization load the modules to list on the dropdown menu */
-   ngOnInit() {
-      this.ms.loadModules();
-   }
-
-   ngDoCheck() {
-      if (this.nodes.length == 0) {
-         this.nodes = this.ms.nodes;
-      }
-   }
-
-   /**
-    * Lists the available content for module input
-    * @param {Module} module
-    */
-   listContent(module: Module) {
-      if (null == this.moduleContents.get(module)) {
-         this.contentVisible.set(module, false);
-         let filter: Filter = new Filter(
-            null, null, [module.id]
-         );
-         this.cs.filterContent(filter).subscribe(
-            (response) => {
-               if (response != null) {
-                  this.parseContentResponse(response, module);
-               } else {
-                  this.toastr.error('Response was null');
-               }
+    /** Map of Visibility status of each Module */
+    contentVisible: Map<Module, boolean> = new Map<Module, boolean>();
```

Comment:
Is a map necessary here? Can multiple Modules be visible at once?

Reply:
Made the map into a variable

Reply:
What then is the difference between the active module and the contentVisible and selModule?

Reply:
+contentActive

Reply:
Apparently nothing. I squashed contentVisible and selModule and changed the references to them to activeModule

Reply:
contentActive is the mapping of what piece of content is considered active in each module. That should be a map since it tracks which piece is active even if you close that module